scripts/verify-jenkins.sh

```diff
@@ -0,0 +1,230 @@
+#       location and all of the plugin exists and plugin version verifications will fail
+#
+
+set -o pipefail
+
+# Default Github Jenkins base url
+GITHUB_JENKINS_BASE_URL='https://raw.githubusercontent.com/openshift/jenkins'
+
+# Default file/folder locations
+PLUGINS_DIR=${PLUGINS_DIR:-/opt/openshift/plugins}
+BUNDLE_PLUGINS_LOCATION=${BUNDLE_PLUGINS_LOCATION:-/opt/openshift/bundle-plugins.txt}
+JENKINS_WAR_LOCATION=${JENKINS_LOCATION:-/usr/lib/jenkins/jenkins.war}
+JENKINS_VERSION_LOCATION=${JENKINS_VERSION_LOCATION:-/opt/openshift/jenkins-version.txt}
+JENKINS_START_SCRIPT_LOCATION=/usr/libexec/s2i/run
+COMMIT_JENKINS_VERSION_LOCATION=/tmp/jenkins-version.txt
+COMMIT_BUNDLE_PLUGINS_LOCATION=/tmp/bundle-plugins.txt
+
+# Location of the log file for failed verifications
+FAILED_LOG_LOCATION=/tmp/failed.log
+# Ensure that the log file does not exist from a previous run
+rm -rf $FAILED_LOG_LOCATION
+
+# ERROR_STRINGS
+#
+# An array of error strings to check for in the Jenkins log file
+ERROR_STRINGS=(
+			"Failed to extract the bundled plugin"
+			"Failed Loading plugin"
+			)
+
+# ensure_sha_env_var
+#
+# Ensure that the environment variable SHA is set
+ensure_sha_env_var() {
+	if [ -z "$SHA" ]; then
+		printf "You MUST supply a SHA to verify against\n"
+		printf "Usage: curl https://raw.githubusercontent.com/openshift/jenkins/master/scripts/verify-jenkins.sh | SHA=<commit sha> sh \n"
+		exit 1
+	fi
+}
+
+# The base url for the files that we need from the code repository
+GITHUB_JENKINS_COMMIT_URL="$GITHUB_JENKINS_BASE_URL/$SHA"
+
+# verify_file()
+#
+# verifies that $local_file_path/$file_name exists and that the contents match
+# those of $remote_file_path/$file_name at the provided SHA
+#
+# Example: verify_file "file.txt" "/my/local/path" "/my/remote/path"
+#
+# @param file_name; The file name to use
+# @param local_file_path; The path to the file in the container
+# @param remote_file_path; The path to the file in the github repository
+#
+# @return void
+verify_file(){
+	local file_name=$1
+	local local_file_path="${2}/${file_name}"
+	local remote_file_path="${GITHUB_JENKINS_COMMIT_URL}/${3}/${file_name}"
+
+	# The temporary file where the contents of the remote file are stored
+	local tmp_file_path="/tmp/${file_name}"
+
+	printf "[VERIFYING] %s\n" "${local_file_path}"
+
+	# Download the remote file to a temporary file, verifying the http_status code for the request
+	printf "     - downloading %s ... " "${remote_file_path}"
+	http_status_code=$(curl -s -o "${tmp_file_path}" -w "%{http_code}\n" "${remote_file_path}" 2> /dev/null)
+
+	if [ "$http_status_code" == 200 ]; then
+		printf "completed\n"
+	else
+		printf "failed\n"
+		printf "failed to download %s, http status code: %s" "${remote_file_path}" "${http_status_code}" >> $FAILED_LOG_LOCATION
+	fi
+
+	# Check if the local file exists
+	printf "     - checking if %s exists ... " "${local_file_path}"
+	if test -f "${local_file_path}}"; then
```

Comment:
```suggestion
	if test -f "${local_file_path}"; then
```
The `file exists` check seems to be not working as expected due to extra `}`. For example:

> [VERIFYING] /opt/openshift/jenkins-version.txt
> - downloading https://raw.githubusercontent.com/openshift/jenkins/eec8e193b75fa8225d06c7cf73492fd7d22c5ab8/2/contrib/openshift/jenkins-version.txt ... completed
> - checking if /opt/openshift/jenkins-version.txt exists ... no
> - checking if /opt/openshift/jenkins-version.txt matches https://raw.githubusercontent.com/openshift/jenkins/eec8e193b75fa8225d06c7cf73492fd7d22c5ab8/2/contrib/openshift/jenkins-version.txt ... yes

The above reports that the file  `/opt/openshift/jenkins-version.txt` does not exists, but it was able to compare the content with the `remote_file` successfully & hence reported yes in next check.
And when we perform the ls, could confirm that this file already exists, yet reported to be missing:

> Some errors were encountered:
> /opt/openshift/jenkins-version.txt does not exist but should

Output of ls:

> bash-4.4$ ls /opt/openshift/jenkins-version.txt
> /opt/openshift/jenkins-version.txt
> bash-4.4$ cat /opt/openshift/jenkins-version.txt
> 2.361.4

Reply:
Fixed